### PR TITLE
feature: better profanity filter

### PR DESCRIPTION
**Pull request by AnshMasand**
## Description

Implement a profanity detecting transformer for the Human Protocol recording oracle. 

## Summary of changes

We have changed the curses file to download the transformer locally the first time the endpoint is accessed.
It returns a boolean value if the model is confident that the sentence contains hate speech/profanity/offensive language. 

## How test the changes

N/A

## Related issues
We are running into issues while developing, more info to follow. 

## Operational checklist

- [ ] All new functionality is covered by tests
- [ ] Any related documentation has been changed or added

### CODE REVIEW

File: packages/examples/fortune/recording-oracle/src/plugins/config.ts
Comment (simulacrum6) [inline packages/examples/fortune/recording-oracle/src/plugins/config.ts:1]: Why was this file changed?

File: packages/examples/fortune/recording-oracle/src/plugins/curses.ts [hunk 1]
```diff
@@ -1,465 +1,14 @@
...
-    for (let i = 0; i < curses.length; i++) {
-      const expression = new RegExp(curses[i], 'gi');
-      if (expression.test(text)) {
-        return true;
-      }
-    }
+  async isProfane(text: string) {
+    const result = await (await pipe)(text);
+    const profanity = result[0].find((res: { label: string; }) => res.label === "Profanity_detected");
+    return (profanity && profanity.score > 0.5);
```
Comment (simulacrum6) [inline packages/examples/fortune/recording-oracle/src/plugins/curses.ts:11]: The score check is not necessary, pipe returns only the most likely label anyways, so score will always be > 0.5, or am I overlooking something?

**Comment by simulacrum6** [comment]
@eugenvoronov At some point, we might want to look into the performance overhead of using transformer models. It might not matter much for the Fortune example, but processing one item at a time is fairly inefficient with neural networks. Usually, it is better to process them in batches.

File: packages/examples/fortune/recording-oracle/src/plugins/config.ts
Comment (simulacrum6) [inline packages/examples/fortune/recording-oracle/src/plugins/config.ts:1]: Did you add your access token to the [HF_ACCESS_TOKEN](https://huggingface.co/docs/transformers.js/guides/private) environment variable, or did you make the model public?

File: packages/examples/fortune/recording-oracle/src/plugins/curses.ts [hunk 2]
```diff
@@ -1,465 +1,14 @@
 import fp from 'fastify-plugin';
 import { FastifyPluginAsync } from 'fastify';
+import { pipeline } from '@xenova/transformers';
 
-const curses = [
```
Comment (simulacrum6) [inline packages/examples/fortune/recording-oracle/src/plugins/curses.ts:4]: I think we agreed on having a two-stage approach, where both a wordlist and the model prediction is checked. You mentioned an extended / improved list of cursewords.

Comment (simulacrum6) [reply packages/examples/fortune/recording-oracle/src/plugins/curses.ts:4]: Alright, misunderstanding then.

File: packages/examples/fortune/recording-oracle/src/plugins/config.ts
Comment (simulacrum6) [reply packages/examples/fortune/recording-oracle/src/plugins/config.ts:1]: @eugenvoronov

File: packages/examples/fortune/recording-oracle/src/plugins/curses.ts (hunk 1 above)
Comment (simulacrum6) [reply packages/examples/fortune/recording-oracle/src/plugins/curses.ts:11]: It's alright. In case we want to add different models, I suppose we'd define a common interface for them anyways.

**Review by eugenvoronov — APPROVED** [review]
